kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/client/PssCheckpoint.java

```diff
@@ -0,0 +1,175 @@
+package dev.responsive.kafka.internal.db.rs3.client;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import java.util.Objects;
+import java.util.UUID;
+
+public class PssCheckpoint {
+  private final UUID storeId;
+  private final int pssId;
+  private final StorageCheckpoint checkpoint;
+
+  @JsonCreator
+  public PssCheckpoint(
+      @JsonProperty("storeId") final UUID storeId,
+      @JsonProperty("pssId") final int pssId,
+      @JsonProperty("checkpoint") StorageCheckpoint checkpoint,
+      @JsonProperty("slateDbStorageCheckpoint")
```

Comment:
I'll clean this up in a follow-up. I left it in since I have a bunch of checkpoints lying around that I still want to be able to deserialize, and at the same time would like to start rebasing my snapshot branch on this pr (and later prs) as they get merged to main